x-pack/plugin/esql/qa/testFixtures/src/main/resources/unmapped_fields.csv-spec

```
@@ -0,0 +1,518 @@
+| INSIST client_ip
+| KEEP @timestamp, client_ip
+| SORT @timestamp DESC
+;
+
+@timestamp:date           |  client_ip:unsupported
+2024-10-23T13:55:01.543Z  |  null
+2024-10-23T13:53:55.832Z  |  null
+2024-10-23T13:52:55.015Z  |  null
+2024-10-23T13:51:54.732Z  |  null
+2024-10-23T13:33:34.937Z  |  null
+2024-10-23T12:27:28.948Z  |  null
+2024-10-23T12:15:03.360Z  |  null
+;
+
+fieldIsMappedToNonKeywordWithCastSingleIndex
+required_capability: unmapped_fields
+FROM partial_mapping_sample_data
+| INSIST client_ip
+| EVAL client_ip = client_ip :: IP
+| KEEP @timestamp, client_ip
+| SORT @timestamp DESC
+;
+
+@timestamp:datetime      | client_ip:ip
+2024-10-23T13:55:01.543Z | 173.21.3.15
+2024-10-23T13:53:55.832Z | 173.21.3.15
+2024-10-23T13:52:55.015Z | 173.21.3.15
+2024-10-23T13:51:54.732Z | 173.21.3.15
+2024-10-23T13:33:34.937Z | 173.21.0.5
+2024-10-23T12:27:28.948Z | 173.21.2.113
+2024-10-23T12:15:03.360Z | 173.21.2.162
+;
+
+fieldIsMappedToNonKeywordWithCastToDifferentTypeSingleIndex
+required_capability: unmapped_fields
+FROM partial_mapping_sample_data
+| INSIST @timestamp
+| EVAL @timestamp = @timestamp :: LONG
+| KEEP @timestamp
+| SORT @timestamp DESC
+;
+
+@timestamp:long
+1729691701543
+1729691635832
+1729691575015
+1729691514732
+1729690414937
+1729686448948
+1729685703360
+;
+
+fieldIsMappedToKeywordSingleIndex
+required_capability: unmapped_fields
+FROM partial_mapping_sample_data
+| INSIST message
+| KEEP @timestamp, message
+| SORT @timestamp DESC
+;
+
+@timestamp:datetime      |  message:keyword
+2024-10-23T13:55:01.543Z |  Connected to 10.1.0.1!
+2024-10-23T13:53:55.832Z |  Connection error?
+2024-10-23T13:52:55.015Z |  Connection error?
+2024-10-23T13:51:54.732Z |  Connection error?
+2024-10-23T13:33:34.937Z |  42
+2024-10-23T12:27:28.948Z |  Connected to 10.1.0.2!
+2024-10-23T12:15:03.360Z |  Connected to 10.1.0.3!
+;
+
+fieldDoesNotExistSingleIndex
+required_capability: unmapped_fields
+FROM partial_mapping_sample_data
+| INSIST foo
```

Having a query like `FROM partial_mapping_sample_data | eval x = 1 | INSIST client_ip` and getting an error message is kind of unexpected:
- the error message doesn't simply say what's wrong, ie `insist` must come after `from` _somehow_: `"line 1:49: INSIST command can only be applied on top of a source"`. Maybe you can improve this with `.... on top of a source command (from, row etc)`.
  - But does this work with `row`? I tested `row a = 1 | INSIST client_ip` and I get `class org.elasticsearch.xpack.esql.plan.logical.Row cannot be cast to class org.elasticsearch.xpack.esql.plan.logical.EsRelation`
- that `eval x = 1` has nothing to do with `insist` and also doesn't change the query in any way

---

* Sure, I'll change the message.
* I'll fix the bug with `ROW`.
* You're right that the `EVAL x = 1` has nothing to do with `INSIST`, but it makes the code **much** easier if I don't have to consider all the various places `INSIST` can appear in :) As this syntax is a work in progress, I'd rather keep this requirement as is for now, and revisit it later if needed.

---

On further thought, should we really allow for `ROW`? It's almost certainly a mistake, as the `INSIST` command is a noop in that case. I think it would make more sense to forbid erroneous code than to silently ignore it, or even warn about it. WDYT?